src/lifecycle.js

```diff
@@ -346,16 +343,36 @@ function triggerDetached (target, component) {
 }
 
 /**
- * Triggers the entire element lifecycle if it's not being ignored.
+ * Creates a tree-walker pre-configured for walking a tree for only nodes that
+ * Skate cares about.
  *
- * @param {Element} target The component element.
- * @param {Object} component The component data.
+ * @param {DOMElement} element The element to walk.
  *
- * @returns {undefined}
+ * @returns {TreeWalker}
  */
-function triggerLifecycle (target, component) {
-  triggerCreated(target, component);
-  triggerAttached(target, component);
+function createTreeWalker (element) {
```

Comment:
`createTreeWalker` → `createElementTreeWalker` maybe? Since you're using just `NodeFilter.SHOW_ELEMENT`.